README.md

```diff
@@ -15,28 +15,6 @@
 npm install usb-detection
```

Comment:
> Is there a scenario where we can merge `node-usb-detection` into `node-usb` (or at least the windows implementation) to cover the deficiencies in libusb and align efforts? I'd love to see the JavaScript polling removed and native events used instead.
>
> *-- @thegecko, https://github.com/MadLittleMods/node-usb-detection/pull/127#issuecomment-978322845*

The reason I started using this project [6 years ago](https://github.com/MadLittleMods/node-usb-detection/pull/9) was because it was the only project that actually worked on Windows (and the other OS's). Although not my use case, it also seems like a lot of people use this with Electron so it would be good to have a compatible option there too.

The second criteria is documentation which looks to be available on https://github.com/node-usb/node-usb#usbdetection

A third criteria is it actually being able to be built and installed easily. This means prebuilds or ideally no C++ code necessary if that was even possible.

If those can be addressed(which some already are), then I could see this being deprecated in favor of [`usb`](https://www.npmjs.com/package/usb).

Reply:
@thegecko (repost that in this thread)

Reply:
@MadLittleMods I believe with the `v2.x.x` release of `node-usb`, all of your criteria are now met

Reply:
@thegecko Tested and seems to work. I think we can move forward with deprecating as I'm running into a wall trying to get prebuilds to work anyway (https://github.com/MadLittleMods/node-usb-detection/pull/165#discussion_r823362667).

```js
const { usb, getDeviceList} = require('usb');

const devices = getDeviceList();
console.log('devices', devices);

usb.on('attach', function(device) { console.log('attach', device) });

usb.on('detach', function(device) { console.log('detach', device) });
```

---

To note some differences and preemptively answer some peoples questions, how do we get `deviceName`, `manufacturer`, `serialNumber` when using the `usb` package? Maybe I'm just not understanding how to get a `USBDevice` which seems to have those properties as described in the readme and somehow mixing the legacy API.

`usb-detection`
```
{
    locationId: 336592896,
    vendorId: 6353,
    productId: 20193,
    deviceName: 'Pixel 4a',
    manufacturer: 'Google',
    serialNumber: 'xxx',
    deviceAddress: 7
  }
```

`usb`
```
{
  busNumber: 20,
  deviceAddress: 5,
  deviceDescriptor: {
    bLength: 18,
    bDescriptorType: 1,
    bcdUSB: 512,
    bDeviceClass: 0,
    bDeviceSubClass: 0,
    bDeviceProtocol: 0,
    bMaxPacketSize0: 64,
    idVendor: 6353,
    idProduct: 20193,
    bcdDevice: 1088,
    iManufacturer: 1,
    iProduct: 2,
    iSerialNumber: 3,
    bNumConfigurations: 1
  },
  portNumbers: [ 1 ]
}
```

Reply:
The legacy API returns the raw USB data which doesn't undertake further blocking calls to obtain the strings (e.g. serial number, device name, etc.).

You can get these in a manner of ways:

- Use the legacy device you got from `getDeviceList()` or your legacy event and call [getStringDescriptor](https://github.com/node-usb/node-usb#getstringdescriptorindex-callbackerror-data) for each of the string IDs
- [Wrap the legacy device](https://github.com/node-usb/node-usb#turn-legacy-device-into-webusb-compatible-device) into a [webusb](https://wicg.github.io/webusb/#usbdevice) compatible device which pre-populates these strings
- Use the WebUSB interface to [request a device](https://github.com/node-usb/node-usb#use-webusb-approach-to-list-all-devices) or register for events:
```typescript
import { WebUSB } from 'usb';

(async () => {
    const customWebUSB = new WebUSB({
        // Bypass checking for authorised devices
        allowAllDevices: true
    });

    customWebUSB.addEventListener('connect', event => console.log(event.device);
})();
```